socketio/src/asynchronous/client/client.rs

```diff
@@ -516,6 +515,42 @@ mod test {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn socket_io_async_callback() -> Result<()> {
+        // Test whether asynchronous callbacks are fully executed.
+        let url = crate::test::socket_io_server();
+
+        // This synchronization mechanism is used to let the
+        let notify = Arc::new(tokio::sync::Notify::new());
+        let notify_clone = notify.clone();
+
+        let socket = ClientBuilder::new(url)
+            .on("test", move |_, _| {
+                let cl = notify_clone.clone();
+                async move {
+                    sleep(Duration::from_secs(1)).await;
+                    // The async callback should be awaited and not aborted.
+                    // Thus, the notification should be called.
+                    cl.notify_one();
+                }
+                .boxed()
+            })
+            .connect()
+            .await?;
+
+        let payload = json!({"token": 123_i32});
+        let result = socket
+            .emit("test", Payload::String(payload.to_string()))
+            .await;
+
+        assert!(result.is_ok());
+        // If the timeout did not trigger, the async callback was fully executed.
+        let timeout = timeout(Duration::from_secs(5), notify.notified()).await;
```

Comment:
Does `Notify` have a build in timeout or a timeout option? If yes it might be nicer to use it, otherwise this is good!

Reply:
I didn't find one in the [documentation](https://docs.rs/tokio/latest/tokio/sync/struct.Notify.html>).

Reply:
Alright, then this is the was to go. Thanks for checking :)